rpa-home/modules/ROOT/pages/automation-usecase-legacysystemdataextraction-extractfromscreen.adoc

```diff
@@ -0,0 +1,97 @@
+# Interacting with a Screen of a Legacy Application
+
+// number the headings to correspond to the numbered list of steps
+:sectnums:
+
+With the order ID you can now get the corresponding tracking number from the legacy system:
+
+. Run your legacy application.
+. Search for the word `Order-ID` on the screen, click in the edit field next to it, enter the order ID and click *Search*.
+. If there is a result, copy the tracking number to the activity parameter and set the gateway variable `is_orderid_found` to true. If there is no result, set the gateway variable `is_orderid_found` to false.
+
+image::usecase-legacysystem-screeninteraction.png["The Workflow contains the steps in form of Action Steps. Some of the Action Steps are nested."]
+
+## Run a Legacy Desktop Application
+
+To run the legacy desktop order tracking application, you select the path to the executable file in the *Run program* Action Step. You set the start mode of the application to `Maximized` to open the application in a maximized window. This ensures that the elements of the UI are always in the same position on the screen, which is important because you later use Action Steps that work with screen coordinates. You select `Wait for Window to appear` and specify the window title to ensure that the following Action Steps won't return an error because the window isn't there yet.
+
+image::usecase-legacysystem-screeninteraction-run.png[]
+
+## Enter Text Into a Screen Element and Click a Button
+
+To search for the tracking number, you have to enter the order ID in the search field and click the search button.
+
+To find the search field, you need to find the word `Order ID` next to it on the application screen first. You use the *Search Pattern* Action Step to capture and compare the pattern:
+
+image::usecase-legacysystem-screeninteraction-capturepattern.png["When you capture a pattern, the screen freezes. You draw a box around the pattern and the search area on the frozen screen."]
+
+image::usecase-legacysystem-screeninteraction-searchpattern.png[]
+
+Finding the pattern returns a coordinate that you use in the next step to determine where to input the order ID. You use the *Mouse Action* Action Step to click into the search field next to the label `Order ID`:
+
+image:usecase-legacysystem-screeninteraction-mouseaction.png["The coordinates from the found pattern are used in the Relative To field of the Mouse Action Action Step."]
+
+The search field now has the focus, so you can just enter the order ID with a simulated keyboard. Instead of recording key strokes, you use the Activity Parameter `order_id` with the *Enter String* Action Step to copy the Order ID into the field:
```

Comment:
```suggestion
The search field now has the focus, so you can just enter the order ID with a simulated keyboard. Instead of recording keystrokes, you use the Activity Parameter `order_id` with the *Enter String* Action Step to copy the Order ID into the field:
```

Reply:
global: search for "order ID" and ensure proper capitalization.